arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Option.kt

```diff
@@ -812,19 +828,149 @@ fun <A> Iterable<Option<A>>.combineAll(MA: Monoid<A>): Option<A> =
     acc.combine(MA, a)
   }
 
-fun <T> Iterable<T>.firstOrNone(): Option<T> = this.firstOrNull().toOption()
+/**
+ * Returns the first element as [Some(element)][Some], or [None] if the iterable is empty.
+ */
+fun <T> Iterable<T>.firstOrNone(): Option<T> =
```

Comment:
All extensions on `Iterable` are in `Iterable.kt` so they're all grouped under `arrow.core.IterableKt` from Java.